tools/cli/commands/changelog.js

```diff
@@ -817,61 +806,64 @@ async function promptChangelog( argv, needChangelog, types ) {
  * Prompts you for how you want changelogger to run (add to all projects or not, etc).
  *
  * @param {object} argv - the arguments passed.
- * @param {Array} needChangelog - files that need changelogs.
+ * @param {Array} defaultProjects - projects that use the default changelog types.
  * @param {Array} uniqueProjects - projects with unique changelog types.
  * @returns {argv}.
  */
-async function changelogAddPrompt( argv, needChangelog, uniqueProjects ) {
-	const totalProjects = [ ...needChangelog, ...uniqueProjects ];
+async function changelogAddPrompt( argv, defaultProjects, uniqueProjects ) {
+	const totalProjects = [ ...defaultProjects, ...uniqueProjects ];
+	let prompts;
 
-	// If only one project needs a changelog, confirm.
+	/* changelogConfirm: Add changelog for the one project? */
 	if ( totalProjects.length === 1 ) {
-		const response = await inquirer.prompt( {
+		prompts = {
 			type: 'confirm',
 			name: 'changelogConfirm',
 			message: `Add changelog for ${ totalProjects[ 0 ] }?`,
-		} );
-		return response;
-	}
-
-	// If multiple projects need a changelog that can be the same, confirm.
-	if ( uniqueProjects.length === 0 && needChangelog.length > 1 ) {
-		const response = await inquirer.prompt( [
+		};
+	} else if ( defaultProjects.length <= 1 && uniqueProjects.length > 0 ) {
+		/* separateChangelogFiles: Found N projects needing changelogs. Create for each one? */
```

Comment:
😀 The comments from my pseudocode are kind of pointless as code comments, they just repeat what the prompt messages say (and in more stilted language).

A good comment might point out _why_ the first two only have one question (for "separateChangelogFiles") while the other two have two, namely that the "sameChangelog" mode only makes sense if defaultProjects,length > 1.